### ISSUE TITLE
Support multiple Azure organization

### ISSUE THREAD

**Issue by LvffY** [backstage/backstage#10431]
## Feature Suggestion

With my team, we're working with multiple Azure Devops organizations and we'd like to be able to add integration and azure plugins for all of our oranizations in the same backstage app. 

Please let me know if you need more infos or if there is anything that is not clear.

**Comment by awanlin** [comment]
Hi @LvffY, I helped with the Azure DevOps plugin - this shows Builds and Pull Requests as well as has a Pull Request Dashboard. I have some ideas as to how we can support multiple organizations for this but currently I'm pretty busy. If you'd like to contribute this I'd be happy to help review any PRs and do some testing - we use Azure DevOps Server which is on-premise so I can test to make sure things work for it.

As for the integrations side I'll leave that up to the Backstage maintainers but I suspect if you submit a PR to add support for this they would be happy to help review and provide feedback

**Comment by LvffY** [comment]
Hi @awanlin 

Thanks for your quick answer ! :)

Unfortunately I'm data engineer and not a web app developper, so I could not help on this point :(

No soucy about you being busy, but do you have any idea about the time you could look on this ? 

*We're on a POC with backstage with my teammate so not in a hurry*

**Comment by LvffY** [comment]
@awanlin no problem, don't worry :)

May be you could briefly explain your idea and the point where to look if anyone fall into this issue ? (for the Azure DevOps plugin, of course)

**Comment by KariemSoudy** [comment]
@awanlin , this might sound odd, but i can't find any guide to integrate azure devops into backstage, i need to be able to import catalog yaml files from azure devops repos, and also show the PRs and builds in the entity page when i import an entity, and also use templates to create a repo in azure devops,

any help from you would be really appreciated :)

**Comment by awanlin** [comment]
@KariemSoudy, that is odd, all this information is on the Backstage.io website:

I recommend starting with this: https://backstage.io/docs/getting-started/

Here's details on Azure DevOps integration which helps with registering `catalog-info.yaml` files, this can be done using locations: https://backstage.io/docs/integrations/azure/locations or using discovery: https://backstage.io/docs/integrations/azure/discovery

The Azure DevOps plugin (this is found on the plugins page on the Backstage.io website) will show you builds and pull requests: https://github.com/backstage/backstage/blob/master/plugins/azure-devops/README.md 

Once you have that all working then I would look at the Software Templates documentation: https://backstage.io/docs/features/software-templates/software-templates-index

If you have more questions, please log a separate issue for them or post in the discussions area, or on the Backstage Discord server. That will keep this focused on the initial request

**Comment by LvffY** [comment]
@awanlin Is there any update on the multiple azure organization support ? 

This could be related to this  #10402

**Comment by awanlin** [comment]
@LvffY, I have no plans to work on this at all, sorry if I gave you the impression that I did. Feel free to contribute it yourself, I would make time to help test if needed

**Comment by ctoestreich** [comment]
This is something that would be imperative in any large organization.  We are facing this now and wanted to use backstage for consolidation of our orgs.   Could this be accomplished by using proxy config per azure instance similar to this https://roadie.io/backstage/plugins/argo-cd/#support-for-multiple-argocd-instances---option-1

**Comment by awanlin** [comment]
@ctoestreich, unfortunately not, the way the integration is setup doesn't use the proxy and same with the Azure DevOps plugin. Maybe you and @LvffY could work together to contribute this? As I mentioned I'm happy to help test or review PRs, etc.

**Comment by LvffY** [comment]
Up for stale

**Comment by freben** [comment]
This is likely a case that's similar to what's being worked on for AWS. https://github.com/backstage/backstage/pull/14125

**Comment by timhill1989** [comment]
Up for stale

**Comment by JetteBke** [comment]
Is there any plan to implement the support for multiple ADO organizations? It would be a great feature to have!

**Comment by Rugvip** [comment]
@JetteBke nothing that we're aware of from the maintainer side and not somewhere we can spend time, but happy to guide contributions!

**Comment by sanderaernouts** [comment]
@awanlin, can this issue be reopened? I see it was closed as not planned, but since I have a PR open on it that is close to completion I'd say we can close this issue (again) after #18213 is merged :)
